aws-globalaccelerator-endpointgroup/src/main/kotlin/software/amazon/globalaccelerator/endpointgroup/ListHandler.kt

```diff
@@ -0,0 +1,54 @@
+package software.amazon.globalaccelerator.endpointgroup
+
+import com.amazonaws.services.globalaccelerator.model.EndpointGroup
+import com.amazonaws.services.globalaccelerator.model.ListEndpointGroupsRequest
+import com.amazonaws.services.globalaccelerator.model.ListEndpointGroupsResult
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy
+import software.amazon.cloudformation.proxy.HandlerErrorCode
+import software.amazon.cloudformation.proxy.Logger
+import software.amazon.cloudformation.proxy.OperationStatus
+import software.amazon.cloudformation.proxy.ProgressEvent
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest
+import software.amazon.globalaccelerator.endpointgroup.AcceleratorClientBuilder.client
+
+/**
+ * List handler implementation for endpoint group resource.
+ */
+class ListHandler : BaseHandler<CallbackContext?>() {
+    override fun handleRequest(proxy: AmazonWebServicesClientProxy,
+                               request: ResourceHandlerRequest<ResourceModel>,
+                               callbackContext: CallbackContext?,
+                               logger: Logger): ProgressEvent<ResourceModel, CallbackContext?> {
+        logger.debug("List EndpointGroup Request: $request")
+        val agaClient = client
+        val model = request.desiredResourceState
+        val listEndpointGroupsRequest = ListEndpointGroupsRequest().withListenerArn(model.listenerArn).withNextToken(request.nextToken)
+        val listEndpointGroupsResult = proxy.injectCredentialsAndInvoke(listEndpointGroupsRequest, agaClient::listEndpointGroups)
+        val endpointGroupList = convertEndpointGroupList(listEndpointGroupsResult.endpointGroups, model.listenerArn)
+        return ProgressEvent.builder<ResourceModel, CallbackContext>()
+                .status(OperationStatus.SUCCESS)
+                .resourceModels(endpointGroupList)
+                .nextToken(listEndpointGroupsResult.nextToken)
```

Comment:
Just curious, but do the list operations function in a way that they will keep requesting more content until the next token is null?  I ask because I want to confirm OperationStatus.SUCCESS is correct.

Reply:
No it is up to the caller to make another call if there is a nextToken returned in the result.